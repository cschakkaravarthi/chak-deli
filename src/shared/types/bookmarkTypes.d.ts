import Bookmark from '../models/Bookmark.model';
export declare const LOAD_BOOKMARKS = "LOAD_BOOKMARKS";
export declare const FETCH_BOOKMARKS = "FETCH_BOOKMARKS";
export declare const SET_BOOKMARK = "SET_BOOKMARK";
export declare const SAVE_BOOKMARK = "SAVE_BOOKMARK";
export declare const DELETE_BOOKMARK = "DELETE_BOOKMARK";
export declare const REMOVE_BOOKMARK = "REMOVE_BOOKMARK";
export interface LoadBookmarksAction {
    type: typeof LOAD_BOOKMARKS;
    bookmarks: Bookmark[];
}
export interface SetBookmarkAction {
    type: typeof SET_BOOKMARK;
    success: boolean;
}
export interface DeleteBookmarkAction {
    type: typeof DELETE_BOOKMARK;
    success: boolean;
}
export interface BookmarkState {
    bookmarks: Bookmark[];
    success: boolean;
}
//# sourceMappingURL=bookmarkTypes.d.ts.map