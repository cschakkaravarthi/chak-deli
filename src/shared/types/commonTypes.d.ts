export declare const TOGGLE_GLOBAL_TOAST = "TOGGLE_GLOBAL_TOAST";
export interface ToggleGlobalToastAction {
    type: typeof TOGGLE_GLOBAL_TOAST;
    displayToast: boolean;
    toastMessage: string;
    toastError: boolean;
}
export interface CommonState {
    displayToast: boolean;
    toastMessage: string;
    toastError: boolean;
}
//# sourceMappingURL=commonTypes.d.ts.map