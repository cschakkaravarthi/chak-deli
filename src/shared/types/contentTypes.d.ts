import Link from '../models/Link.model';
import Article from '../models/Article.model';
import EventModel from '../models/Event.model';
import { Page } from '../models/Page.model';
import { DocumentSearch } from '../models/Document.model';
import { Department } from '../models/Department.model';
import BoxModel from '../models/Box.model';
export interface TaxonomyTermModel {
    drupal_id: number;
    title: string;
    weight?: number;
    icon?: any;
}
export interface BaseContentModel {
    type: string;
    drupal_id: number;
    title: string;
    summary?: string;
    categories: TaxonomyTermModel[];
    category_ancestors: TaxonomyTermModel[];
    owner: TaxonomyTermModel[];
}
export declare type VariousContentModel = Link | Article | EventModel | Page | DocumentSearch | Department | BoxModel;
export interface VariousContentGroupModel extends TaxonomyTermModel {
    content: VariousContentModel[];
}
export interface ContentStateModel {
    content: ContentCategorizedStorageModel;
    groupedContent: ContentGroupedCategorizedStorageModel;
}
export interface ContentGroupedCategorizedStorageModel {
    [key: string]: VariousContentGroupModel[];
}
export interface ContentCategorizedStorageModel {
    [key: string]: VariousContentModel[];
}
export declare const FETCH_CONTENT = "FETCH_CONTENT";
export declare const LOAD_CONTENT = "LOAD_CONTENT";
export declare const FETCH_GROUPED_CONTENT = "FETCH_GROUPED_CONTENT";
export declare const LOAD_GROUPED_CONTENT = "LOAD_GROUPED_CONTENT";
export interface LoadGroupedContentAction {
    type: typeof LOAD_GROUPED_CONTENT;
    content: VariousContentGroupModel[];
    filterId: string;
}
export interface LoadContentAction {
    type: typeof LOAD_CONTENT;
    content: VariousContentModel[];
    filterId: string;
}
export declare const dummyVariousContentGroup: VariousContentGroupModel;
export declare const dummyVariousContent: VariousContentModel[];
export declare const dummyContentGroupedCategorizedStorage: ContentGroupedCategorizedStorageModel;
//# sourceMappingURL=contentTypes.d.ts.map