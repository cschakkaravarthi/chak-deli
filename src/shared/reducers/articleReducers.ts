import { Reducer } from 'redux';
import {
  ArticleState,
  CLEAR_ARTICLES_LIST,
  FETCH_MORE_ARTICLES,
  LOAD_ARTICLE_BY_ID,
  LOAD_ARTICLES,
  SET_ARTICLE_FILTER_FACET
} from '../types/articleTypes';

export const initialState = {
  articleList: [],
  isLastPage: false,
  facets: {},
  selectedFacet: '0',
  pageCount: 1
};

const articleReducers: Reducer<ArticleState, any> = (
  state = initialState,
  action
) => {
  switch (action.type) {
    case FETCH_MORE_ARTICLES:
      return {
        ...state,
        isLastPage: action.articleList.length === 0,
        articleList: state.articleList.concat(action.articleList),
        facets: action.facets,
        pageCount: state.pageCount + 1
      };
    case LOAD_ARTICLES:
      return {
        ...state,
        articleList: action.articleList
      };
    case LOAD_ARTICLE_BY_ID:
      return {
        ...state,
        articleList: [
          ...state.articleList.filter(article => article.drupal_id !== action.article.drupal_id),
          action.article
        ]
      };
    case SET_ARTICLE_FILTER_FACET:
      return {
        ...state,
        selectedFacet: action.selectedFacet,
        isLastPage: false,
        articleList: [],
        pageCount: 1
      };
    case CLEAR_ARTICLES_LIST:
      return {
        ...state,
        articleList: [],
        pageCount: 1,
        isLastPage: false
      };
    default:
      return state;
  }
};

export default articleReducers;
