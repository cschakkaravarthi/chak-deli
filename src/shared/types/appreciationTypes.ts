import { AppreciationCardNotificationModel as AlertModel } from '../models/Notification.model';
import { AppreciationCardTemplate, AppreciationCardsListGrouped } from '../models/AppreciationCards.model';

export const FETCH_APPRECIATION_DATA = 'FETCH_APPRECIATION_DATA';
export const LOAD_APPRECIATION_ALERT = 'LOAD_APPRECIATION_ALERT';
export const LOAD_APPRECIATION_ALERT_ERROR = 'LOAD_APPRECIATION_ALERT_ERROR';
export const CLEAR_APPRECIATION_REDUCER = 'CLEAR_APPRECIATION_REDUCER';
export const LOAD_APPRECIATION_MANAGER_ALERT = 'LOAD_APPRECIATION_MANAGER_ALERT';
export const SET_NEW_CARD_DETAIL = 'SET_NEW_CARD_DETAIL';
export const FETCH_CARD_TEMPLATES = 'FETCH_CARD_TEMPLATES';
export const CLEAR_NEW_CARD_FORM = 'CLEAR_NEW_CARD_FORM';
export const TOGGLE_SUCCESS_SENT_ALERT = 'TOGGLE_SUCCESS_SENT_ALERT';
export const GET_APPRECIATION_CARDS_LIST_GROUPED = 'GET_APPRECIATION_CARDS_LIST_GROUPED';
export const VIEW_APPRECIATION_CARD = 'VIEW_APPRECIATION_CARD';
export const APPRECIATION_CARD_ID = 'APPRECIATION_CARD_ID';
export const ARCHIVE_ALERT_STATE = 'ARCHIVE_ALERT_STATE';
export const LOAD_CARDS_IN_GROUP = 'LOAD_CARDS_IN_GROUP';

export interface AppreciationState {
  cardId: string;
  cardArchiveSuccess: boolean;
  isFetching: boolean;
  appreciationCardAlert?: AlertModel[];
  appreciationCardAlertError?: boolean;
  appreciationManagerCardAlert?: AlertModel[];
  newAppreciationCardForm: NewAppreciationCardForm;
  cardTemplates: AppreciationCardTemplate[];
  showSentSucessAlert: boolean;
  appreciationCardsListGrouped?: AppreciationCardsListGrouped;
  cardToView?: AlertModel;
  displayCardView: boolean;
}

export interface NewAppreciationCardForm {
  senderName?: string;
  templateId?: string;
  cardMessage?: string;
  recipientEmail?: string;
  recipientLastName?: string;
  recipientFirstName?: string;
  recipientSupervisorEmail?: string;
  managerName?: string;
}

export interface ArchiveAlertState {
  cardArchiveSuccess: boolean;
  type: typeof ARCHIVE_ALERT_STATE;
}

export interface SetArchiveAlert {
  cardId: string;
  type: typeof APPRECIATION_CARD_ID;
}

export interface FetchRequest {
  isFetching: boolean;
  type: typeof FETCH_APPRECIATION_DATA;
}

export interface LoadAppreciationAlertAction {
  isFetching: boolean;
  type: typeof LOAD_APPRECIATION_ALERT;
  appreciationCardAlert: AlertModel[];
}

export interface LoadAppreciationManagerAlertAction {
  isFetching: boolean;
  type: typeof LOAD_APPRECIATION_MANAGER_ALERT;
  appreciationManagerCardAlert: AlertModel[];
}

export interface AppreciationCardAlertError {
  isFetching: boolean;
  type: typeof LOAD_APPRECIATION_ALERT_ERROR;
  appreciationCardAlertError?: boolean;
}

export interface ClearAprreciationAction {
  type: typeof CLEAR_APPRECIATION_REDUCER;
}

export interface SetNewCardDetailAction {
  type: typeof SET_NEW_CARD_DETAIL;
  payload: NewAppreciationCardForm;
}

export interface SetDetailGroupAction {
  type: typeof SET_NEW_CARD_DETAIL;
  payload: NewAppreciationCardForm;
}

export interface FetchAppreciationCardTemplatesAction {
  type: typeof FETCH_CARD_TEMPLATES;
  cardTemplates: AppreciationCardTemplate[];
}

export interface ClearNewCardFormAction {
  type: typeof CLEAR_NEW_CARD_FORM;
}

export interface ToggleSentSuccessAlert {
  type: typeof TOGGLE_SUCCESS_SENT_ALERT;
  showSentSucessAlert: boolean;
}

export interface GetAppreciationCardsListGroupedAction {
  type: typeof GET_APPRECIATION_CARDS_LIST_GROUPED;
  appreciationCardsListGrouped: AppreciationCardsListGrouped;
}

export interface ViewAppreciationCardAction {
  type: typeof VIEW_APPRECIATION_CARD;
  displayCardView: boolean;
  cardToView?: AlertModel;
}

export interface AddAppreciationCardsToGroupAction {
  type: typeof LOAD_CARDS_IN_GROUP;
  appreciationCards: AlertModel[];
  cardsType: string;
}
