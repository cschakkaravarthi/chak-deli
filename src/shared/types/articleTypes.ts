import Article from '../models/Article.model';
import Facet, { FacetProperty } from '../models/Facet.model';
import { VariousContentModel } from './contentTypes';

export const LOAD_ARTICLES = 'LOAD_ARTICLES';
export const LOAD_ARTICLE_BY_ID = 'LOAD_ARTICLE_BY_ID';
export const FETCH_MORE_ARTICLES = 'LOAD_MORE_ARTICLES';
export const SET_ARTICLE_FILTER_FACET = 'SET_ARTICLE_FILTER_FACET';
export const CLEAR_ARTICLES_LIST = 'CLEAR_ARTICLES_LIST';
export const LIKE_ARTICLE_TOGGLE = 'LIKE_ARTICLE_TOGGLE';

export interface ArticleState {
  article?: Article;
  isLastPage: boolean;
  articleList: Article[];
  facets: Facet;
  selectedFacet: string;
  pageCount: number;
}

export interface LoadArticlesAction {
  type: typeof LOAD_ARTICLES;
  articleList: Article[];
}

export interface LoadArticleByIdAction {
  type: typeof LOAD_ARTICLE_BY_ID;
  article: VariousContentModel;
}
export interface LoadMoreArticlesAction {
  type: typeof FETCH_MORE_ARTICLES;
  articleList?: Article[];
  facets?: Facet;
}
export interface SetFilterFacetAction {
  type: typeof SET_ARTICLE_FILTER_FACET;
  selectedFacet: FacetProperty;
}

export interface ClearArticlesListAction {
  type: typeof CLEAR_ARTICLES_LIST;
}

export interface LikeArticleToggleAction {
  type: typeof LIKE_ARTICLE_TOGGLE;
  articleId: number;
  likedByUser: boolean;
}
