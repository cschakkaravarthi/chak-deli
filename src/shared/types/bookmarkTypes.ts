import Bookmark from '../models/Bookmark.model';

export const LOAD_BOOKMARKS = 'LOAD_BOOKMARKS';
export const FETCH_BOOKMARKS = 'FETCH_BOOKMARKS';
export const SET_BOOKMARK = 'SET_BOOKMARK';
export const SAVE_BOOKMARK = 'SAVE_BOOKMARK';
export const DELETE_BOOKMARK = 'DELETE_BOOKMARK';
export const REMOVE_BOOKMARK = 'REMOVE_BOOKMARK';

export interface LoadBookmarksAction {
  type: typeof LOAD_BOOKMARKS;
  bookmarks: Bookmark[];
}

export interface SetBookmarkAction {
  type: typeof SET_BOOKMARK;
  success: boolean;
}

export interface DeleteBookmarkAction {
  type: typeof DELETE_BOOKMARK;
  success: boolean;
}

export interface BookmarkState {
  bookmarks: Bookmark[];
  success: boolean;
}
