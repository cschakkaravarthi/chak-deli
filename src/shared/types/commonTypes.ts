import RejectModal from './../models/RejectModal';

export const CURRENT_USER_NAME = 'CURRENT_USER_NAME';
export const TOGGLE_GLOBAL_TOAST = 'TOGGLE_GLOBAL_TOAST';
export const SET_REJECT_MODAL_TEXTAREA = 'SET_REJECT_MODAL_TEXTAREA';
export const SET_REJECT_MODAL_ID_ISSHOW = 'SET_REJECT_MODAL_ID_ISSHOW';

export interface ToggleGlobalToastAction {
  type: typeof TOGGLE_GLOBAL_TOAST;
  displayToast: boolean;
  toastMessage: string;
  toastError: boolean;
}

export type CurrentUser = {
  family_name: string;
  given_name: string;
};

export interface SetCurrentUsername {
  type: typeof CURRENT_USER_NAME;
  currentUsername: CurrentUser;
}

export interface CommonState {
  currentUsername: CurrentUser;
  displayToast: boolean;
  toastMessage: string;
  toastError: boolean;
  idAndIsShow: RejectModal;
  textarea: string;
}

export interface SetRejectModalIdIsShow {
  type: typeof SET_REJECT_MODAL_ID_ISSHOW;
  idAndIsShow: RejectModal;
}

export interface SetRejectModalTextarea {
  type: typeof SET_REJECT_MODAL_TEXTAREA;
  textarea: string;
}
