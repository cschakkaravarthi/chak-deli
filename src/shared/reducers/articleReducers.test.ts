import {
  LOAD_ARTICLE_BY_ID,
  FETCH_MORE_ARTICLES
} from './../types/articleTypes';
import articleReducers, { initialState } from './articleReducers';
import { LOAD_ARTICLES } from '../types/articleTypes';
import { dummyArticles, dummyArticle } from '../models/Article.model';
import { mockedFacets } from '../models/Facet.model';

describe('articleReducers', () => {
  it('Handles default case', () => {
    const state = articleReducers(undefined, {});
    expect(state).toEqual(initialState);
  });

  it('Handles loadArticles action', () => {
    const action = { type: LOAD_ARTICLES, articleList: dummyArticles };
    const state = articleReducers(undefined, action);
    expect(state).toMatchSnapshot();
  });

  it('Handles loadArticleById action', () => {
    const action = { type: LOAD_ARTICLE_BY_ID, article: dummyArticle };
    const state = articleReducers(undefined, action);
    expect(state).toMatchSnapshot();
  });

  it('Handles loadMoreArticles action', () => {
    const action = {
      type: FETCH_MORE_ARTICLES,
      articleList: dummyArticles.concat(dummyArticles),
      facets: mockedFacets
    };
    const state = articleReducers(undefined, action);
    expect(state).toMatchSnapshot();
  });
});
