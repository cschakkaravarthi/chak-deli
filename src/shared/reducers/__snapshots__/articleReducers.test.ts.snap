// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articleReducers Handles loadArticleById action 1`] = `
Object {
  "articleList": Array [
    Object {
      "articleCategory": Array [
        Object {
          "drupal_id": 45,
          "icon": null,
          "title": "Industry News",
          "weight": 1,
        },
      ],
      "body_full": "<p><strong>An Article!</strong>&nbsp;About&nbsp;<em>music</em>! This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.</p>",
      "categories": Array [
        Object {
          "drupal_id": 1,
          "icon": "test.png",
          "title": "test",
          "weight": 1,
        },
      ],
      "category_ancestors": Array [],
      "created": 1568727365,
      "drupal_id": 1,
      "image_uri": Object {
        "source": "undefined",
        "umgc_banner": "undefined",
        "umgc_embedded": "undefined",
        "umgc_featured": "undefined",
        "umgc_thumbnail": "undefined",
      },
      "likedByUser": true,
      "likesCount": 5,
      "owner": Array [
        Object {
          "drupal_id": 1,
          "icon": null,
          "title": "Corporate Communications",
          "weight": 0,
        },
      ],
      "summary": "summary content",
      "tags": Array [
        Object {
          "drupal_id": 1,
          "title": "Company news article 6",
        },
      ],
      "title": "Test Article",
      "type": "article",
    },
  ],
  "facets": Object {},
  "isLastPage": false,
  "pageCount": 1,
  "selectedFacet": "0",
}
`;

exports[`articleReducers Handles loadArticles action 1`] = `
Object {
  "articleList": Array [
    Object {
      "articleCategory": Array [
        Object {
          "drupal_id": 45,
          "icon": null,
          "title": "Industry News",
          "weight": 1,
        },
      ],
      "body_full": "<p><strong>An Article!</strong> About <em>music</em>! This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content.</p><p><a href=\\"https://www.google.com/search?q=music\\">Here is a link</a></p><ul><li>Bullet point!</li><li>Another one!</li></ul><h2>A second-level heading!!</h2><h3>And a third!!</h3><wombat>This tag is unknown!</wombat><script>alert(\\"This content should be stripped!\\")</script>",
      "categories": Array [],
      "category_ancestors": Array [],
      "created": 1561047294,
      "drupal_id": 1,
      "image_uri": Object {
        "source": "undefined",
        "umgc_banner": "undefined",
        "umgc_embedded": "undefined",
        "umgc_featured": "undefined",
        "umgc_thumbnail": "undefined",
      },
      "likedByUser": false,
      "likesCount": 5,
      "owner": Array [
        Object {
          "drupal_id": 1,
          "icon": null,
          "title": "Corporate Communications",
          "weight": 0,
        },
      ],
      "summary": "An article about music.",
      "tags": Array [
        Object {
          "drupal_id": 90,
          "title": "Corporate Communications",
        },
      ],
      "title": "Zeppelin",
      "type": "article",
    },
    Object {
      "articleCategory": Array [
        Object {
          "drupal_id": 45,
          "icon": null,
          "title": "Industry News",
          "weight": 1,
        },
      ],
      "body_full": "<p><strong>An Article!</strong> About <em>music</em>! This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content.</p><p><a href=\\"https://www.google.com/search?q=music\\">Here is a link</a></p><img src=\\"https://assets.nrdc.org/sites/default/files/styles/full_content--retina/public/media-uploads/duplaix_shortclawedotter3.jpg?itok=mF0khuw-\\" title=\\"Embedded Video\\" /><ul><li>Bullet point!</li><li>Another one!</li></ul><h2>A second-level heading!!</h2><h3>And a third!!</h3><wombat>This tag is unknown!</wombat><script>alert(\\"This content should be stripped!\\")</script>",
      "categories": Array [],
      "category_ancestors": Array [],
      "created": 1561047294,
      "drupal_id": 2,
      "image_uri": Object {
        "source": "undefined",
        "umgc_banner": "undefined",
        "umgc_embedded": "undefined",
        "umgc_featured": "undefined",
        "umgc_thumbnail": "undefined",
      },
      "likedByUser": false,
      "likesCount": 5,
      "owner": Array [
        Object {
          "drupal_id": 1,
          "icon": null,
          "title": "Corporate Communications",
          "weight": 0,
        },
      ],
      "summary": "An article about music.",
      "tags": Array [
        Object {
          "drupal_id": 90,
          "title": "Corporate Communications",
        },
      ],
      "title": "Floyd",
      "type": "article",
    },
    Object {
      "articleCategory": Array [
        Object {
          "drupal_id": 45,
          "icon": null,
          "title": "Industry News",
          "weight": 1,
        },
      ],
      "body_full": "<p><strong>An Article!</strong> About <em>music</em>! This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content.</p><p><a href=\\"https://www.google.com/search?q=music\\">Here is a link</a></p><img src=\\"https://assets.nrdc.org/sites/default/files/styles/full_content--retina/public/media-uploads/duplaix_shortclawedotter3.jpg?itok=mF0khuw-\\" title=\\"Embedded Video\\" /><ul><li>Bullet point!</li><li>Another one!</li></ul><h2>A second-level heading!!</h2><h3>And a third!!</h3><wombat>This tag is unknown!</wombat><script>alert(\\"This content should be stripped!\\")</script>",
      "categories": Array [],
      "category_ancestors": Array [],
      "created": 1561047294,
      "drupal_id": 3,
      "image_uri": Object {
        "source": "undefined",
        "umgc_banner": "undefined",
        "umgc_embedded": "undefined",
        "umgc_featured": "undefined",
        "umgc_thumbnail": "undefined",
      },
      "likedByUser": false,
      "likesCount": 0,
      "owner": Array [
        Object {
          "drupal_id": 1,
          "icon": null,
          "title": "Corporate Communications",
          "weight": 0,
        },
      ],
      "summary": "An article about music.",
      "tags": Array [
        Object {
          "drupal_id": 90,
          "title": "Corporate Communications",
        },
      ],
      "title": "Hendrix",
      "type": "article",
    },
  ],
  "facets": Object {},
  "isLastPage": false,
  "pageCount": 1,
  "selectedFacet": "0",
}
`;

exports[`articleReducers Handles loadMoreArticles action 1`] = `
Object {
  "articleList": Array [
    Object {
      "articleCategory": Array [
        Object {
          "drupal_id": 45,
          "icon": null,
          "title": "Industry News",
          "weight": 1,
        },
      ],
      "body_full": "<p><strong>An Article!</strong> About <em>music</em>! This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content.</p><p><a href=\\"https://www.google.com/search?q=music\\">Here is a link</a></p><ul><li>Bullet point!</li><li>Another one!</li></ul><h2>A second-level heading!!</h2><h3>And a third!!</h3><wombat>This tag is unknown!</wombat><script>alert(\\"This content should be stripped!\\")</script>",
      "categories": Array [],
      "category_ancestors": Array [],
      "created": 1561047294,
      "drupal_id": 1,
      "image_uri": Object {
        "source": "undefined",
        "umgc_banner": "undefined",
        "umgc_embedded": "undefined",
        "umgc_featured": "undefined",
        "umgc_thumbnail": "undefined",
      },
      "likedByUser": false,
      "likesCount": 5,
      "owner": Array [
        Object {
          "drupal_id": 1,
          "icon": null,
          "title": "Corporate Communications",
          "weight": 0,
        },
      ],
      "summary": "An article about music.",
      "tags": Array [
        Object {
          "drupal_id": 90,
          "title": "Corporate Communications",
        },
      ],
      "title": "Zeppelin",
      "type": "article",
    },
    Object {
      "articleCategory": Array [
        Object {
          "drupal_id": 45,
          "icon": null,
          "title": "Industry News",
          "weight": 1,
        },
      ],
      "body_full": "<p><strong>An Article!</strong> About <em>music</em>! This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content.</p><p><a href=\\"https://www.google.com/search?q=music\\">Here is a link</a></p><img src=\\"https://assets.nrdc.org/sites/default/files/styles/full_content--retina/public/media-uploads/duplaix_shortclawedotter3.jpg?itok=mF0khuw-\\" title=\\"Embedded Video\\" /><ul><li>Bullet point!</li><li>Another one!</li></ul><h2>A second-level heading!!</h2><h3>And a third!!</h3><wombat>This tag is unknown!</wombat><script>alert(\\"This content should be stripped!\\")</script>",
      "categories": Array [],
      "category_ancestors": Array [],
      "created": 1561047294,
      "drupal_id": 2,
      "image_uri": Object {
        "source": "undefined",
        "umgc_banner": "undefined",
        "umgc_embedded": "undefined",
        "umgc_featured": "undefined",
        "umgc_thumbnail": "undefined",
      },
      "likedByUser": false,
      "likesCount": 5,
      "owner": Array [
        Object {
          "drupal_id": 1,
          "icon": null,
          "title": "Corporate Communications",
          "weight": 0,
        },
      ],
      "summary": "An article about music.",
      "tags": Array [
        Object {
          "drupal_id": 90,
          "title": "Corporate Communications",
        },
      ],
      "title": "Floyd",
      "type": "article",
    },
    Object {
      "articleCategory": Array [
        Object {
          "drupal_id": 45,
          "icon": null,
          "title": "Industry News",
          "weight": 1,
        },
      ],
      "body_full": "<p><strong>An Article!</strong> About <em>music</em>! This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content.</p><p><a href=\\"https://www.google.com/search?q=music\\">Here is a link</a></p><img src=\\"https://assets.nrdc.org/sites/default/files/styles/full_content--retina/public/media-uploads/duplaix_shortclawedotter3.jpg?itok=mF0khuw-\\" title=\\"Embedded Video\\" /><ul><li>Bullet point!</li><li>Another one!</li></ul><h2>A second-level heading!!</h2><h3>And a third!!</h3><wombat>This tag is unknown!</wombat><script>alert(\\"This content should be stripped!\\")</script>",
      "categories": Array [],
      "category_ancestors": Array [],
      "created": 1561047294,
      "drupal_id": 3,
      "image_uri": Object {
        "source": "undefined",
        "umgc_banner": "undefined",
        "umgc_embedded": "undefined",
        "umgc_featured": "undefined",
        "umgc_thumbnail": "undefined",
      },
      "likedByUser": false,
      "likesCount": 0,
      "owner": Array [
        Object {
          "drupal_id": 1,
          "icon": null,
          "title": "Corporate Communications",
          "weight": 0,
        },
      ],
      "summary": "An article about music.",
      "tags": Array [
        Object {
          "drupal_id": 90,
          "title": "Corporate Communications",
        },
      ],
      "title": "Hendrix",
      "type": "article",
    },
    Object {
      "articleCategory": Array [
        Object {
          "drupal_id": 45,
          "icon": null,
          "title": "Industry News",
          "weight": 1,
        },
      ],
      "body_full": "<p><strong>An Article!</strong> About <em>music</em>! This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content.</p><p><a href=\\"https://www.google.com/search?q=music\\">Here is a link</a></p><ul><li>Bullet point!</li><li>Another one!</li></ul><h2>A second-level heading!!</h2><h3>And a third!!</h3><wombat>This tag is unknown!</wombat><script>alert(\\"This content should be stripped!\\")</script>",
      "categories": Array [],
      "category_ancestors": Array [],
      "created": 1561047294,
      "drupal_id": 1,
      "image_uri": Object {
        "source": "undefined",
        "umgc_banner": "undefined",
        "umgc_embedded": "undefined",
        "umgc_featured": "undefined",
        "umgc_thumbnail": "undefined",
      },
      "likedByUser": false,
      "likesCount": 5,
      "owner": Array [
        Object {
          "drupal_id": 1,
          "icon": null,
          "title": "Corporate Communications",
          "weight": 0,
        },
      ],
      "summary": "An article about music.",
      "tags": Array [
        Object {
          "drupal_id": 90,
          "title": "Corporate Communications",
        },
      ],
      "title": "Zeppelin",
      "type": "article",
    },
    Object {
      "articleCategory": Array [
        Object {
          "drupal_id": 45,
          "icon": null,
          "title": "Industry News",
          "weight": 1,
        },
      ],
      "body_full": "<p><strong>An Article!</strong> About <em>music</em>! This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content.</p><p><a href=\\"https://www.google.com/search?q=music\\">Here is a link</a></p><img src=\\"https://assets.nrdc.org/sites/default/files/styles/full_content--retina/public/media-uploads/duplaix_shortclawedotter3.jpg?itok=mF0khuw-\\" title=\\"Embedded Video\\" /><ul><li>Bullet point!</li><li>Another one!</li></ul><h2>A second-level heading!!</h2><h3>And a third!!</h3><wombat>This tag is unknown!</wombat><script>alert(\\"This content should be stripped!\\")</script>",
      "categories": Array [],
      "category_ancestors": Array [],
      "created": 1561047294,
      "drupal_id": 2,
      "image_uri": Object {
        "source": "undefined",
        "umgc_banner": "undefined",
        "umgc_embedded": "undefined",
        "umgc_featured": "undefined",
        "umgc_thumbnail": "undefined",
      },
      "likedByUser": false,
      "likesCount": 5,
      "owner": Array [
        Object {
          "drupal_id": 1,
          "icon": null,
          "title": "Corporate Communications",
          "weight": 0,
        },
      ],
      "summary": "An article about music.",
      "tags": Array [
        Object {
          "drupal_id": 90,
          "title": "Corporate Communications",
        },
      ],
      "title": "Floyd",
      "type": "article",
    },
    Object {
      "articleCategory": Array [
        Object {
          "drupal_id": 45,
          "icon": null,
          "title": "Industry News",
          "weight": 1,
        },
      ],
      "body_full": "<p><strong>An Article!</strong> About <em>music</em>! This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content.</p><p><a href=\\"https://www.google.com/search?q=music\\">Here is a link</a></p><img src=\\"https://assets.nrdc.org/sites/default/files/styles/full_content--retina/public/media-uploads/duplaix_shortclawedotter3.jpg?itok=mF0khuw-\\" title=\\"Embedded Video\\" /><ul><li>Bullet point!</li><li>Another one!</li></ul><h2>A second-level heading!!</h2><h3>And a third!!</h3><wombat>This tag is unknown!</wombat><script>alert(\\"This content should be stripped!\\")</script>",
      "categories": Array [],
      "category_ancestors": Array [],
      "created": 1561047294,
      "drupal_id": 3,
      "image_uri": Object {
        "source": "undefined",
        "umgc_banner": "undefined",
        "umgc_embedded": "undefined",
        "umgc_featured": "undefined",
        "umgc_thumbnail": "undefined",
      },
      "likedByUser": false,
      "likesCount": 0,
      "owner": Array [
        Object {
          "drupal_id": 1,
          "icon": null,
          "title": "Corporate Communications",
          "weight": 0,
        },
      ],
      "summary": "An article about music.",
      "tags": Array [
        Object {
          "drupal_id": 90,
          "title": "Corporate Communications",
        },
      ],
      "title": "Hendrix",
      "type": "article",
    },
  ],
  "facets": Object {
    "tags": Array [
      Object {
        "drupal_id": 1,
        "icon": "string",
        "title": "string",
        "weight": 1,
      },
    ],
  },
  "isLastPage": false,
  "pageCount": 2,
  "selectedFacet": "0",
}
`;
